core/src/fnc/type.rs

```diff
@@ -95,8 +95,19 @@ pub fn table((val,): (Value,)) -> Result<Value, Error> {
 }
 
 pub fn thing((arg1, arg2): (Value, Option<Value>)) -> Result<Value, Error> {
-	Ok(if let Some(arg2) = arg2 {
-		Value::Thing(Thing {
+	match (arg1, arg2) {
+		// Empty table name
+		(Value::Strand(arg1), _) if arg1.is_empty() => Err(Error::TbInvalid {
+			value: arg1.to_string(),
```

Comment:
You may want to use `arg1.as_string()` (rather than `arg1.to_string()`), avoiding an allocation.

Reply:
Thanks! Good spot :D